common/viewer.cpp

```diff
@@ -1583,6 +1583,132 @@ namespace rs2
                 }
             }
 
+            //switch( stream_mv.profile.stream_type() )
+            {
+                static std::vector< std::pair< ImColor, bool > > colors =
+                {
+                    { ImColor( 1.f, 1.f, 1.f, 1.f ), false },  // the default color
+                    { ImColor( 1.f, 0.f, 0.f, 1.f ), false },
+                    { ImColor( 0.f, 1.f, 0.f, 1.f ), false },
+                    { ImColor( 0.f, 0.f, 1.f, 1.f ), false },
+                    { ImColor( 1.f, 0.f, 1.f, 1.f ), false },
+                    { ImColor( 1.f, 1.f, 0.f, 1.f ), false },
+                };
+                typedef size_t ColorIdx;
+                static std::map< size_t, ColorIdx > id2color;
+
+                auto get_color = [&]( object_in_frame const & object ) -> ImColor
+                {
+                    ColorIdx & color = id2color[object.id];
+                    if( color < int( colors.size() ) )
+                    {
+                        if( color > 0 )
+                            return colors[color].first;
```

Comment:
same

Reply:
Do we have a color for the ID?
yes -> return it
find next available color; found it?
yes -> mark it as used; return it
no -> we have more objects than colors; use the the default!